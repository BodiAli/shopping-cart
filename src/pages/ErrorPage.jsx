import { Link } from "react-router-dom";

function ErrorPage() {
  return (
    <>
      <h1>Error Occured</h1>
      <Link to="/">Head Back!</Link>
    </>
  );
}

export default ErrorPage;
