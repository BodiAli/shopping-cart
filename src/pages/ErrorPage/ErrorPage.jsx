import { useEffect } from "react";
import { Link, useRouteError } from "react-router-dom";
import styles from "./ErrorPage.module.css";

function ErrorPage() {
  const error = useRouteError();

  useEffect(() => {
    document.title = "Error";

    return () => {
      document.title = "GameVault";
    };
  });

  return (
    <div className={styles.container}>
      <h1>Looks like you are lost!</h1>
      {error.data && <p>{error.data}</p>}
      <p>{error.statusText || error.message}</p>
      <Link to="/">Head Back!</Link>
    </div>
  );
}

export default ErrorPage;
