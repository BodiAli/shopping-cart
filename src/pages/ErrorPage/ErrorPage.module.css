.container {
  min-height: 100vh;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 50px;
  background: linear-gradient(270deg, var(--green), #7e7c7c);
  background-size: 400% 400%;

  animation: gradient-animation 7s ease infinite;
}

@keyframes gradient-animation {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

.container > h1 {
  font-size: 4rem;
}

.container > p {
  font-size: 2rem;
}

.container > a {
  font-size: 2rem;
  color: black;
}
