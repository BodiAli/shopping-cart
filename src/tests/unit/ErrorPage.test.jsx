import { render, screen } from "@testing-library/react";
import { createMemoryRouter, RouterProvider, useRouteError } from "react-router-dom";
import { vi } from "vitest";
import ErrorPage from "../../pages/ErrorPage/ErrorPage";

vi.mock("react-router-dom", async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    useRouteError: vi.fn(() => new Error("Unexpected error occurred")),
  };
});

const router = createMemoryRouter([
  {
    path: "/",
    element: <ErrorPage />,
  },
]);
const wrapper = () => <RouterProvider router={router} />;

describe("Error page", () => {
  test("should render correct elements", () => {
    const { container } = render(<ErrorPage />, { wrapper });
    expect(container).toMatchSnapshot();
  });

  test("should change document title", () => {
    render(<ErrorPage />, { wrapper });
    expect(document.title).toBe("Error");
  });

  test("should render correct heading", () => {
    render(<ErrorPage />, { wrapper });
    expect(screen.getByRole("heading")).toHaveTextContent(/looks like you are lost/i);
  });

  test("should render error message", () => {
    render(<ErrorPage />, { wrapper });
    expect(screen.getByRole("paragraph")).toHaveTextContent("Unexpected error occurred");
  });

  test("should render error data if provided", () => {
    useRouteError.mockReturnValueOnce({ data: "Some detailed error data" });
    render(<ErrorPage />, { wrapper });
    expect(screen.getByText("Some detailed error data")).toBeInTheDocument();
  });

  test("should render a link to head back home", () => {
    render(<ErrorPage />, { wrapper });
    const linkElement = screen.getByRole("link", { name: /head back/i });
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute("href", "/");
  });
});
