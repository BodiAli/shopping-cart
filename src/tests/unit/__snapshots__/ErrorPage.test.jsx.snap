// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error page > should render correct elements 1`] = `
<div>
  <div
    class="_container_6adc08"
  >
    <h1>
      Looks like you are lost!
    </h1>
    <p>
      Unexpected error occurred
    </p>
    <a
      href="/"
    >
      Head Back!
    </a>
  </div>
</div>
`;
